src/interface/params.hpp

```diff
@@ -41,15 +42,15 @@ class Params {
   void Add(const std::string &key, T value) {
     PARTHENON_REQUIRE_THROWS(!(hasKey(key)), "Key " + key + " already exists");
     myParams_[key] = std::unique_ptr<Params::base_t>(new object_t<T>(value));
-    myTypes_[key] = std::string(typeid(value).name());
+    myTypes_.insert(make_pair(key, std::type_index(typeid(value))));
```

Comment:
This seems like an improvement! But out of curiosity, why did it need to be changed now?

Reply:
Two reasons:

1. I changed the `myTypes_` map to hold `std::type_index` instead of `std::string` in order to be more sure of some type checking. This change likely wasn't needed for what I was trying to accomplish, but I imagine comparing `std::type_index` is safer than comparing strings.
2. The `[]` operator for maps (I think just for assignment) requires the type to have a default constructor. However, `std::type_index` doesn't have a default constructor since there isn't a default type. Hence using `insert` instead.

Reply:
Could you use emplace instead instead of insert.

Reply:
?